File: sample/preview.cpp
#pragma once
#include <iostream>

#define MARCO

// 2-D vector class

namespace maths
{
class vec2
{
  public:
    vec2(double x = 0, double y = 0)
    : x(x),
      y(y)
    {
      std::cout << "created a 2-D vector" << std::endl;
    }

    vec2 operator+(const vec2 &other) const noexcept
    {
      return vec2(x + other.x, y + other.y);
    }

    [[nodiscard]] constexpr double length_squared() noexcept
    {
      return x * x + y * y;
    }

    double x, y;
};
}

File: sample/preview.py
class Vec2:
    def __init__(self, x=0, y=0):
        self.x = x
        self.y = y
        print("created a 2-D vector\n")

    def __add__(self, other):
        return Vec2(self.x + other.x, self.y + other.y)

    def length_squared(self):
        return self.x ** 2 + self.y ** 2